subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -1110,376 +644,465 @@ static void fota_evt_handler(struct azure_fota_event *fota_evt)
-	}
-
-	param.message.topic.topic.size = len;
+	switch (msg->topic.type) {
+	case AZURE_IOT_HUB_TOPIC_EVENT: {
+		az_iot_message_properties az_properties;
+		size_t property_count = (size_t)msg->topic.property_count;
+		// TODO: Properties are already allocated by the caller, possible to use that buffer
+		//	 instead?
+		char property_buf[CONFIG_AZURE_IOT_HUB_MSG_PROPERTY_BUFFER_SIZE] = {0};
 
-	LOG_DBG("Publishing to topic: %s",
-		log_strdup(param.message.topic.topic.utf8));
+		if (property_count > 0) {
+			az_span prop_span = AZ_SPAN_FROM_BUFFER(property_buf);
 
-	return mqtt_publish(&client, &param);
-}
+			LOG_DBG("Message property count: %d", property_count);
 
-int azure_iot_hub_dps_reset(void)
-{
-	return dps_reset();
-}
+			err = az_iot_message_properties_init(&az_properties, prop_span, 0);
+			if (az_result_failed(err)) {
+				LOG_ERR("Failed to initialize properties, az error: 0x%08x", err);
+				return -EFAULT;
+			}
+			err = msg_properties_add(&az_properties, msg->topic.properties,
+						 property_count);
+			if (err) {
+				LOG_ERR("Failed to add properties, az error: %d", err);
 
-static void azure_iot_hub_run(void)
-{
-	int ret;
-	struct pollfd fds[1];
-start:
-	k_sem_take(&connection_poll_sem, K_FOREVER);
-
-	fds[0].events = POLLIN;
-	fds[0].fd = conn_config.socket;
-
-	while (true) {
-		ret = poll(fds, ARRAY_SIZE(fds),
-			   azure_iot_hub_keepalive_time_left());
-
-		/* If poll returns 0 the timeout has expired. */
-		if (ret == 0) {
-			ret = azure_iot_hub_ping();
-			/* -EAGAIN indicates it is not time to ping; try later;
-			 * otherwise, connection was closed due to NAT timeout.
-			 */
-			if (ret && (ret != -EAGAIN)) {
-				LOG_ERR("Cloud MQTT keepalive ping failed: %d", ret);
-				break;
+				return -EMSGSIZE;
 			}
-			continue;
 		}
 
-		if ((fds[0].revents & POLLIN) == POLLIN) {
-			ret = azure_iot_hub_input();
-			if (ret) {
-				LOG_ERR("Cloud MQTT input error: %d", ret);
-				if (ret == -ENOTCONN) {
-					break;
-				}
-			}
+		err = az_iot_hub_client_telemetry_get_publish_topic(
+			&iot_hub_client,
+			(property_count > 0) ? &az_properties : NULL,
+			topic,
+			sizeof(topic),
+			&topic_len);
+		if (az_result_failed(err)) {
+			LOG_ERR("Failed to get telemetry topic, az error: 0x%08x", err);
 
-			/* The connection might have changed during
-			 * call to azure_iot_hub_input(), as MQTT callbacks
-			 * are called from this context, which again allow
-			 * connect()/disconnect() functions to be called.
-			 * Switching from DPS to IoT hub is an example of this.
-			 * Therefore, the fd to poll must be updated to
-			 * account for the event that it has changed.
-			 */
-			fds[0].fd = conn_config.socket;
-
-			if (connection_state_verify(STATE_INIT)) {
-				/* If connection state is set to STATE_INIT at
-				 * this point we know that the socket has
-				 * been closed and we can break out of poll.
-				 */
-				LOG_DBG("The socket is already closed");
-				break;
+			if (err == 0x80010002) {
```

Comment:
Is there some symbol that could be used for the error code? Comparing to values like this is quite error prone.

Reply:
Used AZ_ERROR_NOT_ENOUGH_SPACE instead.